### PR TITLE
Improve `Debug` implementation for `Frame` and `Report`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Since we support getting the type of a `Frame`, this can be used for better printing

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201893074552404/1201481007343178/f) _(internal)_

## 🔍 What does this change?

- Improve `Debug` output for `Frame` and `Report`
- Add a new example to the crate-level docs, which shows the new output better

## 📜 Does this require a change to the docs?

- An example was added, that shows these changes

## 📹 Demo

Output of the following code:
```rust
let content = fs::read_to_string(path)
    .report()
    .change_context(ParseConfigError::new())
    .attach(Suggestion("Use a file you can read next time!"))
    .attach_lazy(|| format!("Could not read file {path:?}"))?;
```
**Before:**
```
Could not read file "config.json"
             at examples/parse_config.rs:45:10

Caused by:
   0: Use a file you can read next time!
             at examples/parse_config.rs:44:10
   1: Could not parse configuration file
             at examples/parse_config.rs:43:10
   2: No such file or directory (os error 2)
             at examples/parse_config.rs:42:10
```

**Now:**:
```
Could not parse configuration file
             at examples/parse_config.rs:43:10
      - Could not read file "config.json"
      - Use a file you can read next time!

Caused by:
   0: No such file or directory (os error 2)
             at examples/parse_config.rs:42:10
```

Output of the above code when using `Frame as Debug`:
```rust
for frame in report.frames() {
    println!("{frame:?}")
}
```

**Before:**
```
Frame { object: "Could not read file \"config.json\"", location: Location { file: "examples/parse_config.rs", line: 45, col: 10 } }
Frame { object: Suggestion("Use a file you can read next time!"), location: Location { file: "examples/parse_config.rs", line: 44, col: 10 } }
Frame { object: ParseConfigError, location: Location { file: "examples/parse_config.rs", line: 43, col: 10 } }
Frame { object: Os { code: 2, kind: NotFound, message: "No such file or directory" }, location: Location { file: "examples/parse_config.rs", line: 42, col: 10 } }

```

**Now:**
```
Frame { attachment: "Could not read file \"config.json\"", location: Location { file: "examples/parse_config.rs", line: 45, col: 10 } }
Frame { attachment: Suggestion("Use a file you can read next time!"), location: Location { file: "examples/parse_config.rs", line: 44, col: 10 } }
Frame { context: ParseConfigError, location: Location { file: "examples/parse_config.rs", line: 43, col: 10 } }
Frame { context: Os { code: 2, kind: NotFound, message: "No such file or directory" }, location: Location { file: "examples/parse_config.rs", line: 42, col: 10 } }
```

### CODE REVIEW

File: packages/libs/error-stack/src/lib.rs [hunk 1]
```diff
@@ -79,83 +79,80 @@
...
+//!             eprintln!("Suggestion: {suggestion}");
+//!         }
+//!     }
 //! }
-//! # assert_eq!(fake_main().unwrap_err().frames().count(), 3);
-//! # Ok(()) }
 //! ```
 //!
-//! which will produce an error and prints it
+//! which will produce an error and will output somethink like
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/src/lib.rs:130]:
```suggestion
//! which will produce an error and will output something like
```

File: packages/libs/error-stack/src/frame.rs [hunk 2]
```diff
@@ -200,9 +200,13 @@ impl Provider for Frame {
 
 impl fmt::Debug for Frame {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/src/frame.rs:202]: Can you also attach an example of what this alternate formatting looks like when debugging a Report?

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]

File: packages/libs/error-stack/src/frame.rs (hunk 2 above)
Comment (TimDiekmann) [reply packages/libs/error-stack/src/frame.rs:202]: Updated the description to show this output as well

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]
Happy to approve but lets wait for upstream branch to merge

**Pull request merged**